tuna/driver.py

```diff
@@ -0,0 +1,107 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+"""Module that encapsulates the DB representation of a Driver cmd"""
+
+from typing import List, Union, Dict
+from abc import ABC, abstractmethod
+from tuna.utils.logger import setup_logger
+from tuna.miopen.db.miopen_tables import ConvolutionConfig
+
+LOGGER = setup_logger('driver_base')
+
+
+class DriverBase(ABC):
+  """Represents db tables based on ConfigType"""
+
+  @abstractmethod
+  def parse_fdb_key(self, line: str):
+    """Overloaded method.Defined in conv&bn driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def parse_row(self, db_obj: ConvolutionConfig):
+    """Overloaded method.Defined in conv&bn driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def set_cmd(self, data_type: str):
+    """Overloaded method.Defined in conv&bn driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def config_set_defaults(self):
+    """Overloaded method.Defined in conv&bn driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def compose_weight_t(self):
+    """Overloaded method.Defined in conv&br driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @staticmethod
+  @abstractmethod
+  def test_skip_arg(tok1: str):
+    """Overloaded method.Defined in conv&br driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @staticmethod
+  @abstractmethod
+  def get_params(tok1: str):
+    """Overloaded method.Defined in conv&br driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @staticmethod
+  @abstractmethod
+  def get_check_valid(tok1: str, tok2: Union[str, int]):
+    """Overloaded method.Defined in conv&br driver child class"""
+    raise NotImplementedError("Not implemented")
+
+  @staticmethod
+  @abstractmethod
+  def get_common_cols() -> List[str]:
+    """Returns common MIOpenDriver command line args"""
+    return ['wall', 'time', 'iter', 'verify']
+
+  def to_dict(self) -> Dict[str, Union[str, int]]:
+    """Return class to dictionary"""
+    copy_dict: Dict[str, Union[str, int]] = {}
+    key: str
+    value: Union[int, str]
+    for key, value in vars(self).items():
+      if key == "_cmd":
+        copy_dict["cmd"] = value
+      else:
+        copy_dict[key] = value
+    return copy_dict
+
+  def __eq__(self, other: object) -> bool:
```

Comment:
this needs to be abstract here and implemented in the child classes, how else are you going to compare the members of the child classes in the base class if they are not even defined yet? there is no point in having a base class __eq__ function

Reply:
the above comment still stands, there is no use in putting this here, we will never (famous last words) compare base classes. This needs a 'Not implemented' here and move the concrete function in the child classes